src/Action.php

```diff
@@ -0,0 +1,9 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App;
+
+interface Action
+{
```

Comment:
I would most definitely be in favour of adding `public function handle(Request $request): Response;` here, even if it couples the interface to the framework.
If we were in a proper hexagonal architecture, this would not make a difference as we are already in a UI layer anyway, so coupling is slightly less of a problem.

Reply:
Alright but you lose the ability to convert request parameters, which some of the controllers use. But if we don't want to use that anymore, that's fine.